packages/jellyfish-api-core/__tests__/category/icxorderbook/makeOffer.test.ts

```diff
@@ -0,0 +1,467 @@
+
+    // List the ICX offers for orderTx = createOrderTxId and check
+    const ordersAfterMakeOffer: Record<string, ICXOrderInfo | ICXOfferInfo> = await client.call('icx_listorders', [{ orderTx: createOrderTxId }], 'bignumber')
+    expect(Object.keys(ordersAfterMakeOffer).length).toBe(2) // extra entry for the warning text returned by the RPC atm.
+    expect((ordersAfterMakeOffer as Record<string, ICXOfferInfo>)[makeOfferTxId]).toStrictEqual(
+      {
+        orderTx: createOrderTxId,
+        status: ICXOrderStatus.OPEN,
+        amount: offer.amount,
+        amountInFromAsset: offer.amount.dividedBy(order.orderPrice),
+        ownerAddress: offer.ownerAddress,
+        takerFee: offer.amount.multipliedBy(ICX_TAKERFEE_PER_BTC).multipliedBy(DEX_DFI_PER_BTC_RATE),
+        expireHeight: expect.any(BigNumber)
+      }
+    )
+  })
+
+  it('should make an partial offer to an sell DFI order with input utxos', async () => {
+    const accountDFIBeforeOrder = await container.call('getaccount', [accountDFI, {}, true])
+    // create order - maker
+    const order: ICXOrder = {
+      tokenFrom: idDFI,
+      chainTo: 'BTC',
+      ownerAddress: accountDFI,
+      receivePubkey: '037f9563f30c609b19fd435a19b8bde7d6db703012ba1aba72e9f42a87366d1941',
+      amountFrom: new BigNumber(15),
+      orderPrice: new BigNumber(0.01)
+    }
+    const createOrderResult: ICXGenericResult = await client.icxorderbook.createOrder(order, [])
+    const createOrderTxId = createOrderResult.txid
+    await container.generate(1)
+
+    // list ICX orders
+    const ordersAfterCreateOrder: Record<string, ICXOrderInfo | ICXOfferInfo> = await client.call('icx_listorders', [], 'bignumber')
+    expect((ordersAfterCreateOrder as Record<string, ICXOrderInfo>)[createOrderTxId]).toStrictEqual(
+      {
+        status: ICXOrderStatus.OPEN,
+        type: ICXOrderType.INTERNAL,
+        tokenFrom: symbolDFI,
+        chainTo: order.chainTo,
+        receivePubkey: order.receivePubkey,
+        ownerAddress: order.ownerAddress,
+        amountFrom: order.amountFrom,
+        amountToFill: order.amountFrom,
+        orderPrice: order.orderPrice,
+        amountToFillInToAsset: order.amountFrom.multipliedBy(order.orderPrice),
+        height: expect.any(BigNumber),
+        expireHeight: expect.any(BigNumber)
+      }
+    )
+
+    // make offer to partial amout 10 DFI - taker
+    const offer: ICXOffer = {
+      orderTx: createOrderTxId,
+      amount: new BigNumber(0.1), // 10 DFI = 0.1 BTC
+      ownerAddress: accountBTC
+    }
+
+    // input utxos
+    const utxos = await container.call('listunspent', [1, 9999999, [accountBTC], true])
+    const inputUTXOs: UTXO[] = utxos.map((utxo: UTXO) => {
+      return {
+        txid: utxo.txid,
+        vout: utxo.vout
+      }
+    })
+    const accountBTCBeforeOffer = await container.call('getaccount', [accountBTC, {}, true])
```

Comment:
use `client.call(getaccount)` instead, you are losing precision here

Reply:
hmm. yes, you have a valid point. even though the data and numbers used in these tests will not make any precision problems.
Will update it to retrieve as `lossless`

Reply:
use `bignumber`
and precision can be lost when parsing

take a look at this issue https://github.com/DeFiCh/jellyfish/issues/18 and this implementation https://github.com/DeFiCh/jellyfish/pull/47